torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -2364,6 +2363,26 @@ std::tuple<Tensor, Tensor> atan2_backward(const Tensor& grad, const Tensor& self
             output_mask[0] ? grad * other * recip : Tensor(),
             output_mask[1] ? grad * -self * recip : Tensor() };
 }
+std::tuple<Tensor, Tensor, Tensor> attn_backward(
+    const Tensor& grad_o,
+    const Tensor& grad_a,
+    const Tensor& q,
+    const Tensor& k,
+    const Tensor& v,
+    const Tensor& a) {
+  if (!grad_o.defined() || !grad_a.defined()) {
```

Comment:
I'm not sure this is correct, if only one of them is defined, then the output should be non-zero!

Reply:
yeah that makes sense